iroh-relay/src/server/client.rs

```diff
@@ -146,15 +135,53 @@ impl ClientConn {
     /// Shutdown the reader and writer loops and closes the connection.
     ///
     /// Any shutdown errors will be logged as warnings.
-    pub async fn shutdown(self) {
+    pub(super) async fn shutdown(self) {
         self.done.cancel();
         if let Err(e) = self.handle.await {
             warn!(
-                "error closing actor loop for client connection {:?} {}: {e:?}",
-                self.key, self.conn_num
+                "error closing actor loop for client connection {:?}: {e:?}",
+                self.node_id,
             );
         };
     }
+
+    pub(super) fn send_packet(&self, src: NodeId, data: Bytes) -> Result<(), SendError> {
+        try_send(&self.send_queue, Packet { src, data })
+    }
+
+    pub(super) fn send_disco_packet(&self, src: NodeId, data: Bytes) -> Result<(), SendError> {
+        try_send(&self.disco_send_queue, Packet { src, data })
+    }
+
+    pub(super) fn send_peer_gone(&self, key: NodeId) -> Result<(), SendError> {
+        try_send(&self.peer_gone, key)
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub(super) enum SendError {
+    #[error("packet dropped")]
+    PacketDropped,
+    #[error("sender closed")]
+    SenderClosed,
+}
+
+/// Tries up to `3` times to send a message into the given channel, retrying iff it is full.
+fn try_send<T>(sender: &mpsc::Sender<T>, msg: T) -> Result<(), SendError> {
```

Comment:
This reads... surprising to me. Why would one do this?
Can we add a little more info about this? (Or will this part of the code become obsolete with the PR stack soon anyways?)

Reply:
Ah I see, this code existed before, and it was moved here.